pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -264,3 +292,22 @@ func (c *globalIngressIPController) onDelete(ingressIP *submarinerv1.GlobalIngre
 		return nil
 	}, ingressIP.Status.AllocatedIP)
 }
+
+func (c *globalIngressIPController) ensureInternalServiceExists(ingressIP *submarinerv1.GlobalIngressIP) error {
+	serviceRef := ingressIP.Spec.ServiceRef
+	internalSvc := GetInternalSvcName(serviceRef.Name)
+	key := fmt.Sprintf("%s/%s", ingressIP.Namespace, internalSvc)
+
+	service, exists, err := getService(internalSvc, ingressIP.Namespace, c.services, c.scheme)
+	if err != nil || !exists {
+		return fmt.Errorf("internal service created by Globalnet controller %q does not exist", key)
+	}
+
+	if service.Spec.ExternalIPs[0] != ingressIP.Status.AllocatedIP {
+		_ = deleteService(ingressIP.Namespace, internalSvc, c.services)
+
+		return fmt.Errorf("globalIP assigned to %q does not match with Internal Service ExternalIP", key)
```

Comment:
How would this scenario occur?

Reply:
In an ideal scenario this will not happen, but if user intentionally (or accidentally) modifies the external-ip (aka globalIP) of the submariner-internalService we delete that service and re-create it with correct ip.

Reply:
I guess it doesn't hurt to check here but this would only run on a controller restart which should be very infrequent. Ideally we would handle such erroneous changes in the control loop but I'm not sure it's really worth trying to handle such cases - I'm sure there's other potential erroneous scenarios elsewhere in globalnet et al which we don't handle.

Reply:
I see your POV @tpantelis, thanks. Agree that these are not common scenarios and will only happen if user does something crazy. By handling these scenarios (whenever possible) we are going to save our time while troubleshooting issues particularly when people don't mention the steps they performed (before they land into issues) and expect support to resolve the situation.

I'd like to get some feedback from others on this topic - @nyechiel @skitt @vthapar @aswinsuryan @mkolesnik @mkimuram 

Short summary: Globalnet controller now creates an internal Service for every exported Service. The internal service that is created by Globalnet Controller will be configured with an external-ip which will be nothing but the globalIP assigned to the exported service. Ideally the internal service is owned by Globalnet and users should NOT modify/delete it. But if users modify the external-ip (in some platforms like OCP, this is not possible) or delete the service (possible), should Globalnet controller handle such scenarios by trying to keep the associated GlobalIngressIP in Sync or can we treat this as an configuration issue from the users?

Reply:
> Ideally the internal service is owned by Globalnet and users should NOT modify/delete it. But if users modify the external-ip (in some platforms like OCP, this is not possible) or delete the service (possible), should Globalnet controller handle such scenarios by trying to keep the associated GlobalIngressIP in Sync or can we treat this as an configuration issue from the users (with end result as undefined)?

I understand your concerns and trying to mitigate user mistakes. I think at the very least, where reasonable, we could detect such cases and log error messages. Eg, if an internal service is deleted but the associated `GlobalIngressIP` already exists then log a message. Similarly if the `ExternalIP` of an internal service is modified, log it.

Reply:
I think that @sridhargaddam once shared an idea to create internal Services for exported Services in submariner namespace (maybe for other reasons). By doing so, we would be able to prevent users from accidentally deleting them. However, drawback will be that we need an extra step to sync the endpoints for the service to submariner namespace. Also, we will need to consider how to avoid name collision for the services.

It might be too much just to avoid accidental deletion, but it should address this issue.

Reply:
> prevent users from accidentally deleting them

We can actually do this by adding a finalizer to the internal service. We can also add the `GlobalIngressIP` as the owner so K8s automatically deletes it (after we remove the finalizer).

Reply:
Thanks @tpantelis @mkimuram

We discussed this topic in today's external network meeting and @aswinsuryan also suggested the same approach. So, lets use the finalizer on the internal-service to stop users from deleting the internal services. Also, I'll think about configuring the owner for the internal service with `GlobalIngressIP` so that K8s will take care of it. 

With regards to users modifying the external-ip in the internal-services I'll log a message and will enhance subctl diagnose to detech the inconsistencies. Good points @tpantelis

Reply:
👍 For cleanup on uninstall, we'll have to remove the finalizers.